src/metabase/api/common/internal.clj

```diff
@@ -252,6 +256,45 @@
         route
         (apply vector route arg-types)))))
 
+(defn- ->matching-regex
+  "Note: this is called in a macro context, so it can potentially be passed a symbol that evaluates to a schema."
+  [schema]
+  (let [schema-type (try (mc/type schema)
+                         (catch clojure.lang.ExceptionInfo _
+                           (mc/type (eval schema))))]
+    (cond
+      ;; can use regex directly
+      (= schema-type :re) (first (try (mc/children schema)
+                                      (catch clojure.lang.ExceptionInfo _
+                                        (mc/children (eval schema)))))
+      (= schema-type 'pos-int?) #"[0-9]+"
+      (= schema-type :int) #"-?[0-9]+"
+      (= schema-type 'int?) #"-?[0-9]+"
+      (= schema-type :uuid) u/uuid-regex
+      (= schema-type 'uuid?) u/uuid-regex)))
```

Comment:
(I assume there's a good reason that `'re` and `:pos-int` aren't options? I don't know Malli internals well enough)

Reply:
Really good question. As for `:pos-int` -- it's not actually a schema. It's kind of confusing, but: schemas like `int?` and `pos-int?` are actually different than e.g. `:int` and `[:int]`. Those last 2 are identical. The difference is that the keywordized schemas will use their properties as you'd expect (like: `[:int {:min 20}]`) BUT the symbol ones actually ignore them. For that reason I try to use the kw versions wherever possible (they print better too). 

There is some work going on in malli to improve the dispatching so we don't have to check for both here, but it hasn't landed upstream yet.